Home work 9 1.

zhurlov opened the pull request:
Привет, проверь пожалуйста д.з.
Мне не понятно с пукта 6, последние предложение, а именно:

Требуется создать отдельный класс для запуска приложения и объявить метод main в нем. Объясни подробнее пожалуйста...
Возможно это элементарно, но что то.....)))

zhurlov committed:
Домашняя работа №9. Объекты и классы.

DmitryN10 commented on src/Author.java:
```diff
@@ -0,0 +1,17 @@
+    String name;
+    String surname;
+
+    public String getName() {
```
Написать конструкторы для обоих классов, заполняющие все поля.

DmitryN10 replied:
private String firstName;
    private String lastName;

    public Author(String firstName, String lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

DmitryN10 commented on src/Book.java:
```diff
@@ -0,0 +1,26 @@
+String name;
+int publication;
+
+    public Book(String name, int publication) {
```
У книги должен быть автор.
Надо в конструктор передавать обьект класса Author.

DmitryN10 commented on src/Main.java:
```diff
@@ -1,9 +1,52 @@
 
+////////// Объект Автор.
+        Author vyacheslav = creativeFace("vyacheslav", "grigoriev");
+        //vyacheslav.name = "vyacheslav";
```
Вот так например

Author vyacheslav = creativeFace("vyacheslav", "grigoriev");
Book grigory = new Book("gregory",vyacheslav, 1988);

zhurlov replied:
так то все понятно, но пока не получается. 
 передавать в классе конструктора Book надо объект класса Автор?

DmitryN10 replied:
да

zhurlov committed:
Домашняя работа №9. Объекты и классы.
Привет, проверь пожалуйста д.з.
Мне не понятно с пукта 6, последние предложение, а именно:

Требуется создать отдельный класс для запуска приложения и объявить метод main в нем. Объясни подробнее пожалуйста...
Возможно это элементарно, но что то.....)))

DmitryN10 commented on src/Book.java:
```diff
@@ -0,0 +1,26 @@
+        this.name = name;
+        this. publication = publication;
+
+    }
```
private String name;
    private Author author;
    private int year;

    public Book(String name, Author author, int year) {
        this.name = name;
        this.author = author;
        this.year = year;
    }

zhurlov replied:
Почему у меня так получилось, когда передаю в конструктор объект, я ставлю тип переменной объекта String, просто в твоих подсказках стоит созданный тип Author author, а у меня получилось String author, с созданным типом не получается внести изменения в свойствах класса, тем самым идеа в Маин объекте бук в параметрах видит ошибку.

DmitryN10 replied:
Она видит ошибку тк у тебя сперва создан обьект Book, а потом автор. Должно быть наоборот.
Мы сперва создаем **Author vyacheslav = new Author("vyacheslav", "grigoriev");**

потом его передаем в конструктор Book

Book grigory = new Book("smart student", **vyacheslav**, 1988);

zhurlov replied:
понятно)

DmitryN10 commented on src/Book.java:
```diff
@@ -0,0 +1,22 @@
+public class Book {
+private String name;
+private int publication;
+private String author;
```
private **Author** author;
так долдно быть

DmitryN10 commented on src/Book.java:
```diff
@@ -0,0 +1,22 @@
+private String author;
+
+
+    public Book(String name, String author ,int publication ) {
```
public Book(String name, **Author** author ,int publication ) {

zhurlov committed:
Домашняя работа №9. Объекты и классы.
Поправил, но как то не доволен, как будто не доработал.
1.Почему,у меня так получилось, когда передаю в конструктор объект, я ставлю тип переменной объекта String, просто в твоих подсказках стоит созданный тип Author author, а у меня получилось String author, с созданным типом не получается внести изменения в свойствах класса, тем самым идеа в Маин объекте бук в параметрах видит ошибку.

zhurlov committed:
Домашняя работа №9. Объекты и классы.
Поправил. Вроде,ок.
